src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -42,52 +50,23 @@ export interface SocialMediaLinksProps {
   /** Optional desktop layout. Smaller viewports are always in a column if there are labels and
    * in a row if there are no labels. */
   layout?: LayoutTypes;
-  /** Optional array of social media platform types, urls, and label texts. */
-  linksData?: SocialMediaLinkDataProps[];
-  /** Optional true/false to display names of platforms along with icons.
-   *  NOTE: Labels will NOT be shown with a circular border */
-  showLabels?: boolean;
+  /** Required. Array of social media platform types, urls, and label texts. */
+  linksData: SocialMediaLinkDataProps[];
   /** Optional size: small, medium, or large. */
   size?: SizeType;
 }
 
-/* Accepts an array containing one or more data objects, each representing a social media platform and optional
- * overrides for the label and/or URL.
- *
- * @returns an array of data objects for each requested platform type that includes type, iconName, labelText and
- * url. The labelText and url props will include any values supplied to override the defaults.
- */
-function getLinksData(platforms: SocialMediaLinkDataProps[]) {
-  let allData = [];
-  platforms.forEach((myPlatform) => {
-    // Get the dataset for this platform.
-    let thisPlatformArray = socialMediaDataMap.filter(
-      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
-    );
-
-    // The filter function returns an array of objects, so let's just get the objects out of the array.
-    let thisPlatformData = thisPlatformArray[0];
-
-    // If a url value exists in myPlatform use it, else use the default.
-    let newUrl = myPlatform.url ?? thisPlatformData.url;
-
-    // If a labelText value exists in myPlatform use it, else use the default.
-    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
-
-    let thisObj = {
-      type: thisPlatformData.type,
-      iconName: thisPlatformData.iconName,
-      labelText: newLabelText,
-      url: newUrl,
-    };
-    allData.push(thisObj);
-  }); // end forEach
-
-  return allData;
-}
+type ConditionalSocialMediaLinksProps =
+  | { showLabels?: true; borders?: Exclude<BorderType, "circular"> }
+  | { showLabels?: false; borders?: BorderType };
 
+export type SocialMediaLinksProps = CommonSocialMediaLinksProps &
+  ConditionalSocialMediaLinksProps;
 /**
  * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ * The data passed to the `linksData` prop controls which social media types are displayed,
+ * the order they appear, the url they link to, and the text used for the label of each.
+ * Order is determined by the order of the array.
```

Comment:
You can remove the last sentence, it's covered in the previous sentence.

Reply:
Done.